app/src/main/java/com/eveningoutpost/dexdrip/CalibrationGraph.java

```diff
@@ -114,4 +126,111 @@ public Line getCalibrationsLine(List<Calibration> calibrations, int color) {
         line.setHasLabels(true);
         return line;
     }
+
+
+
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+
+        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(xdrip.getAppContext());
+
+
+        //Just generate the menu in engineering mode
+        if (!prefs.getBoolean("engineering_mode", false)){
+            return false;
+        }
+
+        getMenuInflater().inflate(R.menu.menu_calibrationgraph, menu);
+
+        // Only show elements if there is a calibration to overwrite
+        if(Calibration.lastValid()!= null){
+            menu.findItem(R.id.action_overwrite_intercept).setVisible(true);
+            menu.findItem(R.id.action_overwrite_slope).setVisible(true);
+            menu.findItem(R.id.action_overwrite_calibration_impossible).setVisible(false);
+        }
+
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+
+        switch (item.getItemId()) {
+            case R.id.action_overwrite_intercept:
+                overWriteIntercept();
+                break;
+            case R.id.action_overwrite_slope:
+                overWriteSlope();
+        }
+
+        return true; //consume event
+    }
+
+    private void overWriteIntercept() {
+        final EditText editText = new EditText(this);
+        editText.setInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_FLAG_SIGNED | InputType.TYPE_NUMBER_FLAG_DECIMAL);
+
+        new AlertDialog.Builder(this)
+                .setTitle("Ovewrite Intercept")
+                .setMessage("Overwrite Intercept")
+                .setView(editText)
+                .setPositiveButton("OK", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int which) {
+                        String text = editText.getText().toString();
+                        if (!TextUtils.isEmpty(text)) {
+                            double doubleValue = Double.parseDouble(text);
```

Comment:
For some reason I didn't think `Double.parseDouble(String str)` was locale aware? I made: `JoH.tolerantParseDouble(String str)` to handle issues with different decimal separators. Java documentation seems to say it does support localized. Possibly it becomes an issue if the string is stored and later interpreted..